src/routes/blogs.rs

```diff
@@ -35,7 +37,63 @@ pub fn details(name: String, page: Option<Page>, rockets: PlumeRocket) -> Result
         page.0,
         Page::total(articles_count as i32),
         posts
-    )))
+    ))
+    .into())
+}
+
+#[get("/<custom_domain>?<page>", rank = 2)]
+pub fn custom_details(
+    custom_domain: String,
+    page: Option<Page>,
+    rockets: PlumeRocket,
+) -> Result<RespondOrRedirect, ErrorPage> {
+    let blog = Blog::find_by_host(&rockets, Host::new(custom_domain))?;
+    detail_guts(blog, page, rockets)
+}
+
+#[get("/~/<name>?<page>", rank = 2)]
+pub fn details(
+    name: String,
+    page: Option<Page>,
+    rockets: PlumeRocket,
+) -> Result<RespondOrRedirect, ErrorPage> {
+    let blog = Blog::find_by_fqn(&rockets, &name)?;
+
+    // check this first, and return early
+    // doing this prevents partially moving `blog` into the `match (tuple)`,
+    // which makes it impossible to reuse then.
+    if blog.custom_domain == None {
+        return detail_guts(blog, page, rockets);
+    }
+
+    match (blog.custom_domain, page) {
+        (Some(ref custom_domain), Some(ref page)) => {
+            Ok(Redirect::to(format!("https://{}/?{}", custom_domain, page)).into())
```

Comment:
I think it will not properly format the page: you will get something like `https://doma.in/?2` when you probably want `?page=2`.